src/k8s/pkg/k8sd/api/cluster_tokens.go

```diff
@@ -21,14 +22,16 @@ func postClusterJoinTokens(m *microcluster.MicroCluster, s *state.State, r *http
 		return response.BadRequest(fmt.Errorf("failed to parse request: %w", err))
 	}
 
-	var (
-		token string
-		err   error
-	)
+	hostname, err := utils.CleanHostname(req.Name)
+	if err != nil {
+		return response.BadRequest(fmt.Errorf("invalid hostname %q: %w", req.Name, err))
```

Comment:
Let's add a new error "ErrBadHostname" and add a human-readable error to the CLI.
Currently, we get:
```
ubuntu@peter:~/k8s$ sudo k8s get-join-token bad.
failed to retrieve join token: failed to query endpoint POST /k8sd/cluster/tokens: invalid hostname "bad.": hostname cannot end with a dot (".")
```